app/src/main/java/com/example/footprnt/Discover/Adapters/ListAdapter.java

```diff
@@ -122,12 +122,23 @@ public void bindBusinessDetail(Business business) {
             cardView.setTag(business);
             tvBusinessName.setText(business.getName());
             tvBusinessCategory.setText(business.getCategories().get(0));
-            rating.setRating((float) business.getRating());
-            LayerDrawable stars = (LayerDrawable) rating.getProgressDrawable();
-            stars.getDrawable(2).setColorFilter(Color.parseColor("#659DBD"), PorterDuff.Mode.SRC_ATOP);
-            tvBusinessAddress.setText(business.getAddress().get(0));
+            try {
+                rating.setVisibility(View.VISIBLE);
+                rating.setRating((float) business.getRating());
+                LayerDrawable stars = (LayerDrawable) rating.getProgressDrawable();
+                stars.getDrawable(2).setColorFilter(Color.parseColor("#659DBD"), PorterDuff.Mode.SRC_ATOP);
+            } catch (Exception e) {
```

Comment:
Exception is the most general you can get, what Exception are you trying to catch here? It doesn't look like there's anything you can't prevent by doing null checks